src/wp-admin/includes/theme-previews.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Theme previews using the Site Editor for block themes.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Filters the blog option to return the path for the previewed theme.
+ *
+ * @param string $current_stylesheet The current theme's stylesheet or template path.
+ * @return string The previewed theme's stylesheet or template path.
+ */
+function wp_get_theme_preview_path( $current_stylesheet = null ) {
+	if ( ! current_user_can( 'switch_themes' ) ) {
+		return $current_stylesheet;
+	}
+
+	$preview_stylesheet = ! empty( $_GET['wp_theme_preview'] ) ? $_GET['wp_theme_preview'] : null;
+	$wp_theme           = wp_get_theme( $preview_stylesheet );
+	if ( ! is_wp_error( $wp_theme->errors() ) ) {
+		if ( current_filter() === 'template' ) {
+			$theme_path = $wp_theme->get_template();
+		} else {
+			$theme_path = $wp_theme->get_stylesheet();
+		}
+
+		return sanitize_text_field( $theme_path );
+	}
+
+	return $current_stylesheet;
+}
+
+/**
+ * Adds a middleware to `apiFetch` to set the theme for the preview.
```

Comment:
We need a more detailed docblock for this function ;)
https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#formatting-guidelines

Reply:
```suggestion
 * Adds a middleware to `apiFetch` to set the theme for the preview.
 * This adds a `wp_theme_preview` URL parameter to API requests from the Site Editor, so they also respond as if the theme is set to the value of the parameter.
```